pkg/migration/controllers/migrationschedule.go

```diff
@@ -257,6 +270,31 @@ func (m *MigrationScheduleController) handle(ctx context.Context, migrationSched
 
 		// Start a migration for a policy if required
 		if start {
+			// Reconciling the migrationschedule in remote cluster if there is a change
+			// Doing it here to limit the reconcilation check frequency
+			if remoteMigrSched != nil {
+				updatedHash, err := m.reconcileRemoteMigrationSchedule(migrationSchedule, remoteMigrSched)
+				if err != nil {
+					msg := fmt.Sprintf("Error reconciling remote migration schedule: %v", err)
+					log.MigrationScheduleLog(migrationSchedule).Error(msg)
+					return fmt.Errorf(msg)
+				}
+				if updatedHash != 0 {
```

Comment:
updateHash is always not equal to 0, since the function returns either the sourceHash or destHash. So we will keep updating it on the remote side.

Reply:
updatedHash is an uint64. The variable inside the reconcileRemoteMigrationSchedule function is only getting updated if
1. Remote migration schedule has the hash value in annotation and sourceMigrationScheduleHash != destMigrationScheduleHash
2. If no annotation is present.

Once the hash value gets updated in remote copy, the updatedHash return from that function will be the default value (0).

Reply:
If annotation is present in remote and hash has not updated, we return default value of uninitialized var  updatedHash which is 0. Let's add a comment in reconcileRemoteMigrationSchedule() so it's explicit and doesn't cause confusion.